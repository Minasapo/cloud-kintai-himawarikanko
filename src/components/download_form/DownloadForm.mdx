import {
  Canvas,
  Meta,
  Title,
  Subtitle,
  Description,
  Primary,
  Controls,
  Stories,
  Source,
  Story,
} from "@storybook/blocks";

import * as DownloadFormStories from "./DownloadForm.stories";

<Meta of={DownloadFormStories} />

# DownloadForm

アプリケーション内で勤務表をダウンロードするためのフォームです。\
具体的なユースケースが決まっているものは、ストーリーに記載されています。

<Primary />

# パラメーター

コンポーネント内で使用できるパラメーターです。\
パラメーターを変更することで上にあるコンポーネントが変化します。

<Controls />

# ストーリー

コンポーネントのユースケースを記載しています。\
既存のコンポーネントを使用する際に参考にしてください。
