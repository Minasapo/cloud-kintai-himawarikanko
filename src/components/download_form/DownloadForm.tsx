import { Autocomplete, Box, Button, Stack, TextField } from "@mui/material";
import { DesktopDatePicker } from "@mui/x-date-pickers";
import dayjs from "dayjs";
import { useEffect, useState } from "react";
import { Service, Staff } from "../../client";
import { LoginStaff } from "../staff_list/StaffList";

interface FormState {
  startDate: dayjs.Dayjs | null;
  endDate: dayjs.Dayjs | null;
  aggregateMonth: number;
}

async function fetchStaffs(
  loginStaff: LoginStaff,
  callback: (value: Staff[]) => void
) {
  if (!loginStaff) return;

  const staffs = await Service.getStaffs(loginStaff.id).catch((error) => {
    console.log(error);
    return [] as Staff[];
  });

  callback(staffs);
}

interface AggregateMonth {
  id: number;
  date: dayjs.Dayjs;
}

const initialState: FormState = {
  startDate: null,
  endDate: null,
  aggregateMonth: dayjs().month(),
};

const DownloadForm = ({ loginStaff }: { loginStaff: LoginStaff }) => {
  const [formState, setFormState] = useState<FormState>(initialState);
  const [staffs, setStaffs] = useState<Staff[]>([]);

  const aggregateMonthList = [...Array<number>(12)].map(
    (_, i) =>
      ({
        id: i,
        date: dayjs().month(i),
      } as AggregateMonth)
  );

  useEffect(() => {
    if (!loginStaff) return;

    void fetchStaffs(loginStaff, (value) => setStaffs(value));
  }, [loginStaff]);

  function changeHandler<T>(key: string, value: T) {
    setFormState((prevState) => ({
      ...prevState,
      [key]: value,
    }));
  }

  const handleBulkDownload = () => {
    // 処理なし
  };

  return (
    <Stack
      spacing={4}
      alignItems="center"
      sx={{
        border: 1,
        borderColor: "primary.main",
        borderRadius: "5px",
        pb: 3,
      }}
    >
      <Box
        sx={{
          p: 1,
          width: 1,
          boxSizing: "border-box",
          textAlign: "center",
          backgroundColor: "primary.main",
          color: "primary.contrastText",
          borderRadius: "3px 3px 0 0",
        }}
      >
        ダウンロードオプション
      </Box>
      <Box>
        <Stack
          spacing={3}
          sx={{ display: "inline-block", boxSizing: "border-box" }}
        >
          <Box>
            <Stack direction="row" spacing={1} alignItems="center">
              <Box>
                <DesktopDatePicker
                  label="開始日"
                  format="YYYY/MM/DD"
                  value={formState.startDate}
                  onChange={(event) => changeHandler("startDate", event)}
                  slotProps={{ textField: { variant: "outlined" } }}
                />
              </Box>
              <Box>〜</Box>
              <Box>
                <DesktopDatePicker
                  label="終了日"
                  format="YYYY/MM/DD"
                  value={formState.endDate}
                  onChange={(event) => changeHandler("endDate", event)}
                  slotProps={{ textField: { variant: "outlined" } }}
                />
              </Box>
            </Stack>
          </Box>
          <Box>
            <Autocomplete
              multiple
              limitTags={2}
              id="multiple-limit-tags"
              options={aggregateMonthList}
              getOptionLabel={(option) => `${option.date.format("YYYY年MM月")}`}
              defaultValue={[]}
              renderInput={(params) => (
                <TextField
                  {...params}
                  label="集計対象月"
                  placeholder="集計対象月を選択..."
                />
              )}
              sx={{ width: "500px" }}
            />
          </Box>
          <Box>
            <Autocomplete
              data-testid="autocomplete"
              multiple
              limitTags={2}
              id="multiple-limit-tags"
              options={staffs}
              getOptionLabel={(option) =>
                `${option?.last_name || ""} ${option?.first_name || ""}`
              }
              defaultValue={[]}
              renderInput={(params) => (
                <TextField
                  {...params}
                  label="対象者"
                  placeholder="対象者を入力..."
                />
              )}
              sx={{ width: "500px" }}
            />
          </Box>
        </Stack>
      </Box>
      <Box>
        <Button onClick={handleBulkDownload}>一括ダウンロード</Button>
      </Box>
    </Stack>
  );
};
export default DownloadForm;
