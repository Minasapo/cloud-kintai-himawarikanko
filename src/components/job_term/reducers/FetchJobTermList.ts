import { createAsyncThunk } from "@reduxjs/toolkit";
import dayjs from "dayjs";
import utc from "dayjs/plugin/utc";

import { Configuration, MasterApi } from "../../../api";

import { GetConfiguration } from "../../time_recorder/TimeRecorderAPI";
import {
  mappedOriginWorkPeriodPerMonth,
  OriginWorkPeriodPerMonth,
} from "./FetchWorkPeriodPerMonth";

dayjs.extend(utc);

export enum JobTermStatus {
  NO_REGISTER = "NO_REGISTER",
  NO_SYNCED = "NO_SYNCED",
  SYNCED = "REGISTERED",
}

export interface JobTermList {
  id: number;
  targetMonth: string;
  jobStartDate: string;
  jobEndDate: string;
  status: JobTermStatus;
}

async function fetchWorkPeriodPerMonth(
  conf: Configuration,
  targetDate: string
): Promise<OriginWorkPeriodPerMonth[] | null> {
  const api = new MasterApi(conf);
  const workPeriods = await api
    .getWorkPeriodPerMonthList({
      date: targetDate,
    })
    .then((r) => r)
    .catch(() => null);

  if (!workPeriods) return null;

  try {
    return workPeriods.map((workPeriod) =>
      mappedOriginWorkPeriodPerMonth(workPeriod)
    );
  } catch (error) {
    return null;
  }
}

const fetchJobTermList = createAsyncThunk(
  "jobTermList/fetchJobTermList",
  async ({
    targetDate,
  }: {
    targetDate: dayjs.Dayjs;
  }): Promise<JobTermList[]> => {
    // dayjs()で月初を取得
    const beginOfMonth = dayjs(
      new Date(targetDate.year(), targetDate.month(), 1)
    ).format("YYYYMMDD");

    const workPeriods = await fetchWorkPeriodPerMonth(
      GetConfiguration(),
      beginOfMonth
    )
      .then((r) => r)
      .catch(() => null);

    console.log("workPeriods", workPeriods);

    if (!workPeriods) return [];

    const startYear = targetDate.subtract(1, "year").year();
    const endYear = targetDate.add(1, "year").year();

    const jobTermList: JobTermList[] = [];
    for (let targetYear = startYear; targetYear <= endYear; targetYear += 1) {
      for (let targetMonth = 0; targetMonth < 12; targetMonth += 1) {
        const defaultTargetMonth = dayjs()
          .year(targetYear)
          .month(targetMonth)
          .day(1)
          .format("YYYYMM");

        const defaultJobStartDate = dayjs(
          new Date(targetYear, targetMonth, 1)
        ).format("YYYY-MM-DD");

        const defaultJobEndDate = dayjs(
          new Date(targetYear, targetMonth + 1, 0)
        ).format("YYYY-MM-DD");

        const matchWorkPeriod = workPeriods?.find(
          (workPeriod) => workPeriod.targetMonth === defaultTargetMonth
        );

        const newJobTerm = ((): JobTermList => {
          if (matchWorkPeriod) {
            return {
              id: targetMonth,
              targetMonth: matchWorkPeriod.targetMonth ?? defaultTargetMonth,
              jobStartDate:
                dayjs(matchWorkPeriod.jobStartDate).format("YYYY-MM-DD") ?? "",
              jobEndDate:
                dayjs(matchWorkPeriod.jobEndDate).format("YYYY-MM-DD") ?? "",
              status: JobTermStatus.SYNCED,
            };
          }

          return {
            id: targetMonth,
            targetMonth: defaultTargetMonth,
            jobStartDate: defaultJobStartDate,
            jobEndDate: defaultJobEndDate,
            status: JobTermStatus.NO_REGISTER,
          };
        })();

        jobTermList.push(newJobTerm);
      }
    }

    return jobTermList;
  }
);
export default fetchJobTermList;
