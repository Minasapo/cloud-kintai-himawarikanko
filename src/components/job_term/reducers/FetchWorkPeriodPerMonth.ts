import { createAsyncThunk } from "@reduxjs/toolkit";
import dayjs from "dayjs";
import { MasterApi, WorkPeriodPerMonth } from "../../../api";

export interface OriginWorkPeriodPerMonth
  extends Omit<
    WorkPeriodPerMonth,
    "targetMonth" | "jobStartDate" | "jobEndDate"
  > {
  targetMonth: string | undefined;
  jobStartDate: string | undefined;
  jobEndDate: string | undefined;
}

const isNullDate = (date: Date) => {
  const nullDate = dayjs(0);
  const targetDate = dayjs(date);

  return nullDate.isSame(targetDate);
};

export const mappedOriginWorkPeriodPerMonth = (
  workPeriodPerMonth: WorkPeriodPerMonth
): OriginWorkPeriodPerMonth => ({
  workPeriodPerMonthId: workPeriodPerMonth.workPeriodPerMonthId,
  targetMonth: workPeriodPerMonth.targetMonth,
  jobStartDate: isNullDate(workPeriodPerMonth.jobStartDate)
    ? undefined
    : dayjs(workPeriodPerMonth.jobStartDate).toISOString(),
  jobEndDate: isNullDate(workPeriodPerMonth.jobEndDate)
    ? undefined
    : dayjs(workPeriodPerMonth.jobEndDate).toISOString(),
});

const fetchWorkPeriodPerMonth = createAsyncThunk(
  "workPeriodPerMonth/fetchWorkPeriodPerMonth",
  async ({ date }: { date: string }) => {
    const api = new MasterApi();
    const workPeriods = await api
      .getWorkPeriodPerMonthList({
        date,
      })
      .then((r) => r)
      .catch(() => null);

    if (!workPeriods) return null;

    try {
      const convertedWorkPeriods = workPeriods.map((workPeriod) =>
        mappedOriginWorkPeriodPerMonth(workPeriod)
      );

      return convertedWorkPeriods;
    } catch (error) {
      return null;
    }
  }
);
export default fetchWorkPeriodPerMonth;
