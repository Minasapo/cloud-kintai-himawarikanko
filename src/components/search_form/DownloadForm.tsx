import { useEffect, useState } from "react";

import {
  Autocomplete,
  Box,
  FormControl,
  InputLabel,
  MenuItem,
  Select,
  SelectChangeEvent,
  Stack,
  TextField,
} from "@mui/material";
import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
import { DesktopDatePicker } from "@mui/x-date-pickers/DesktopDatePicker";
import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
import dayjs, { Dayjs } from "dayjs";

import { Staff } from "../../api";
import { useAppDispatch, useAppSelector } from "../../lib/hooks";
import fetchStaffList from "../../lib/staff/FetchStaffList";
import { selectStaffList } from "../../lib/store";
import Button from "../button/Button";

const DownloadForm = () => {
  const staffs = useAppSelector(selectStaffList);
  const [startDate, setStartDate] = useState<Dayjs | null>(dayjs("2023-01-01"));
  const [endDate, setEndDate] = useState<Dayjs | null>(dayjs("2023-01-31"));
  const [aggregateMonth, setAggregateMonth] = useState(202301);
  const [targetStaffs, setTargetStaffs] = useState<Staff[]>([]);
  const dispatch = useAppDispatch();

  const handleStartDateChange = (newValue: Dayjs | null) => {
    setStartDate(newValue);
  };

  const handleEndDateChange = (newValue: Dayjs | null) => {
    setEndDate(newValue);
  };

  const handleAggregateMonthChange = (event: SelectChangeEvent<number>) => {
    setAggregateMonth(event.target.value as number);
  };

  const handleBulkDownload = () => {
    // console.log("bulk download");
  };

  useEffect(() => {
    void dispatch(fetchStaffList());
  }, []);

  useEffect(() => {
    setTargetStaffs(staffs.data);
  }, [staffs]);

  return (
    <Stack
      spacing={4}
      alignItems="center"
      sx={{
        border: 1,
        borderColor: "primary.main",
        borderRadius: "5px",
        pb: 3,
      }}
    >
      <Box
        sx={{
          p: 1,
          width: 1,
          boxSizing: "border-box",
          textAlign: "center",
          backgroundColor: "primary.main",
          color: "primary.contrastText",
          borderRadius: "3px 3px 0 0",
        }}
      >
        ダウンロードオプション
      </Box>
      <Box>
        <LocalizationProvider dateAdapter={AdapterDayjs}>
          <Stack
            spacing={3}
            sx={{ display: "inline-block", boxSizing: "border-box" }}
          >
            <Box>
              <Stack direction="row" spacing={1} alignItems="center">
                <Box>
                  <DesktopDatePicker
                    label="開始日"
                    inputFormat="YYYY/MM/DD"
                    value={startDate}
                    onChange={handleStartDateChange}
                    renderInput={(params) => <TextField {...params} />}
                  />
                </Box>
                <Box>〜</Box>
                <Box>
                  <DesktopDatePicker
                    label="終了日"
                    inputFormat="YYYY/MM/DD"
                    value={endDate}
                    onChange={handleEndDateChange}
                    renderInput={(params) => <TextField {...params} />}
                  />
                </Box>
              </Stack>
            </Box>
            <Box>
              <FormControl fullWidth>
                <InputLabel id="demo-simple-select-label">
                  集計対象月
                </InputLabel>
                <Select
                  labelId="demo-simple-select-label"
                  id="demo-simple-select"
                  value={aggregateMonth}
                  label="集計対象月"
                  onChange={handleAggregateMonthChange}
                >
                  <MenuItem value={202301}>2023/01</MenuItem>
                  <MenuItem value={202302}>2023/02</MenuItem>
                  <MenuItem value={202303}>2023/03</MenuItem>
                  <MenuItem value={202304}>2023/04</MenuItem>
                  <MenuItem value={202305}>2023/05</MenuItem>
                  <MenuItem value={202306}>2023/06</MenuItem>
                  <MenuItem value={202307}>2023/07</MenuItem>
                  <MenuItem value={202308}>2023/08</MenuItem>
                  <MenuItem value={202309}>2023/09</MenuItem>
                  <MenuItem value={202310}>2023/10</MenuItem>
                  <MenuItem value={202311}>2023/11</MenuItem>
                  <MenuItem value={202312}>2023/12</MenuItem>
                </Select>
              </FormControl>
            </Box>
            <Box>
              <Autocomplete
                multiple
                limitTags={2}
                id="multiple-limit-tags"
                options={targetStaffs}
                getOptionLabel={(option) =>
                  `${option.lastName} ${option.firstName}`
                }
                defaultValue={[]}
                renderInput={(params) => (
                  <TextField
                    {...params}
                    label="対象者"
                    placeholder="対象者を入力..."
                  />
                )}
                sx={{ width: "500px" }}
              />
            </Box>
          </Stack>
        </LocalizationProvider>
      </Box>
      <Box>
        <Button label="一括ダウンロード" onClick={handleBulkDownload} />
      </Box>
    </Stack>
  );
};
export default DownloadForm;
