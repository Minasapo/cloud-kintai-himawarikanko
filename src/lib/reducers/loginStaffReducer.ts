import { ActionReducerMapBuilder, createSlice } from "@reduxjs/toolkit";
import { Staff } from "../../api";
import fetchStaff from "../staff/FetchStaff";

export enum StaffStatus {
  NOT_PROCESSING = "NOT_PROCESSING",
  PROCESSING = "PROCESSING",
  DONE = "DONE",
  ERROR = "ERROR",
}

export interface StaffState {
  status: StaffStatus;
  data?: Staff;
}

const initialState: StaffState = {
  status: StaffStatus.NOT_PROCESSING,
  data: undefined,
};

export const getStaffExtraReducers = (
  builder: ActionReducerMapBuilder<StaffState>
) => {
  builder
    .addCase(fetchStaff.pending, (state) => {
      state.status = StaffStatus.PROCESSING;
    })
    .addCase(fetchStaff.fulfilled, (state, action) => {
      state.status = StaffStatus.DONE;
      state.data = action.payload as Staff;
    })
    .addCase(fetchStaff.rejected, (state) => {
      state.status = StaffStatus.ERROR;
    });
};

const loginStaffReducer = createSlice({
  name: "loginStaff",
  initialState,
  reducers: {
    clearStaff(state) {
      state.status = StaffStatus.NOT_PROCESSING;
      state.data = undefined;
    },
  },
  extraReducers: (builder) => getStaffExtraReducers(builder),
});

export default loginStaffReducer.reducer;

export const { clearStaff } = loginStaffReducer.actions;

export const testLoginStaffReducer = (customInitialState: StaffState) =>
  createSlice({
    name: "loginStaff",
    initialState: customInitialState,
    reducers: {
      clearStaff(state) {
        state.status = StaffStatus.NOT_PROCESSING;
        state.data = undefined;
      },
    },
    extraReducers: (builder) => getStaffExtraReducers(builder),
  }).reducer;
