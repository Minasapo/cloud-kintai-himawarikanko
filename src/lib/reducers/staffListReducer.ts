import { ActionReducerMapBuilder, createSlice } from "@reduxjs/toolkit";
import { Staff } from "../../api";
import fetchStaffList from "../staff/FetchStaffList";

export enum StaffListStatus {
  NOT_PROCESSING = "NOT_PROCESSING",
  PROCESSING = "PROCESSING",
  DONE = "DONE",
  ERROR = "ERROR",
}

export interface StaffListState {
  status: StaffListStatus;
  data: Staff[];
  selectedData?: Staff;
}

const initialState: StaffListState = {
  status: StaffListStatus.NOT_PROCESSING,
  data: [],
  selectedData: undefined,
};

export const getStaffListExtraReducers = (
  builder: ActionReducerMapBuilder<StaffListState>
) => {
  builder
    .addCase(fetchStaffList.pending, (state) => {
      state.status = StaffListStatus.PROCESSING;
    })
    .addCase(fetchStaffList.fulfilled, (state, action) => {
      state.status = StaffListStatus.DONE;
      state.data = action.payload as Staff[];
    })
    .addCase(fetchStaffList.rejected, (state) => {
      state.status = StaffListStatus.ERROR;
    });
};

const staffListReducer = createSlice({
  name: "staffList",
  initialState,
  reducers: {
    selectedStaff: (state, action) => {
      state.selectedData = action.payload as Staff;
    },
  },
  extraReducers: (builder) => getStaffListExtraReducers(builder),
});

export default staffListReducer.reducer;

export const { selectedStaff } = staffListReducer.actions;

export const testStaffListReducer = (customInitialState: StaffListState) =>
  createSlice({
    name: "staffList",
    initialState: customInitialState,
    reducers: {},
    extraReducers: (builder) => getStaffListExtraReducers(builder),
  }).reducer;
