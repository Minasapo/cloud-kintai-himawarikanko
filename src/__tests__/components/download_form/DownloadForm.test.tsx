import { configureStore } from "@reduxjs/toolkit";
import { fireEvent, render, within } from "@testing-library/react";
import { Provider } from "react-redux";

import DownloadForm from "../../../components/download_form/DownloadForm";
import { LoginStaffStatus, testLoginStaffReducer } from "../../../lib/reducers/loginStaffReducer";
import { StaffListStatus, testStaffListReducer } from "../../../lib/reducers/staffListReducer";

const mockStore = configureStore({
  reducer: {
    loginStaffReducer: testLoginStaffReducer({
      status: LoginStaffStatus.DONE,
      data: {
        staffId: 999,
        lastName: "田中",
        firstName: "太郎",
        mailAddress: "tanaka@example.com",
        iconPath: "",
        staffRoles: {
          roleId: 2,
          staffId: 999,
          role: {
            roleName: "スタッフ",
          },
        },
      },
    }),
    staffListReducer: testStaffListReducer({
      status: StaffListStatus.DONE,
      data: [],
    }),
  },
});

describe("DownloadForm Component(", () => {
  test("ダウンロードフォームにレイアウト崩れがないか(ビジュアルリグレッション)", () => {
    const { asFragment } = render(
      <Provider store={mockStore}>
        <DownloadForm />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test.skip("選択した値が表示されているか", () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <DownloadForm />
      </Provider>
    );

    const autocomplete = getByTestId("autocomplete");
    const input = within(autocomplete).getByRole("combobox");
    expect(input.innerText).toBeUndefined();

    autocomplete.focus();
    fireEvent.change(input, { target: { value: "田中 太郎" } });
    fireEvent.keyDown(autocomplete, { key: "ArrowDown" });
    fireEvent.keyDown(autocomplete, { key: "Enter" });

    const autocompleteAfter = getByTestId("autocomplete");
    const inputAfter = within(autocompleteAfter).getByRole("combobox");
    expect(inputAfter.innerText).toEqual("田中 太郎");
  });
});
