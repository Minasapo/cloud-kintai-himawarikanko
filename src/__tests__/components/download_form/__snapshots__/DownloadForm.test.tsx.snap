// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadForm Component( ダウンロードフォームにレイアウト崩れがないか(ビジュアルリグレッション) 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-n2r28k-MuiStack-root"
  >
    <div
      class="MuiBox-root css-1doaxl"
    >
      ダウンロードオプション
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiStack-root css-j64l5d-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiStack-root css-1q36rmd-MuiStack-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for=":r0:"
                  id=":r0:-label"
                >
                  開始日
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r0:"
                    inputmode="text"
                    placeholder="YYYY/MM/DD"
                    type="text"
                    value="2022/12/02"
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="Choose date, selected date is Dec 2, 2022"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CalendarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        開始日
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              〜
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for=":r2:"
                  id=":r2:-label"
                >
                  終了日
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r2:"
                    inputmode="text"
                    placeholder="YYYY/MM/DD"
                    type="text"
                    value="2023/01/01"
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="Choose date, selected date is Jan 1, 2023"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CalendarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        終了日
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="demo-simple-select-label"
            >
              集計対象月
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                role="button"
                tabindex="0"
              >
                2023/01
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="202301"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    集計対象月
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1i9x9p3-MuiAutocomplete-root"
            data-testid="autocomplete"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="multiple-limit-tags"
                id="multiple-limit-tags-label"
              >
                対象者
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-invalid="false"
                  autocapitalize="none"
                  autocomplete="off"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  id="multiple-limit-tags"
                  placeholder="対象者を入力..."
                  role="combobox"
                  spellcheck="false"
                  type="text"
                  value=""
                />
                <div
                  class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                >
                  <button
                    aria-label="Open"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                    tabindex="-1"
                    title="Open"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      対象者
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        一括ダウンロード
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
